lua/lvim/bootstrap.lua

```diff
@@ -92,31 +95,36 @@ function M:init(base_dir)
     vim.cmd [[let &packpath = &runtimepath]]
   end
 
-  -- FIXME: currently unreliable in unit-tests
-  if not in_headless then
-    _G.PLENARY_DEBUG = false
-    require("lvim.impatient").setup {
-      path = join_paths(self.cache_dir, "lvim_cache"),
-      enable_profiling = true,
-    }
+  _G.PLENARY_DEBUG = false
+  require("lvim.impatient").setup {
+    path = self.lua_cache_path,
+    enable_profiling = true,
+  }
+  if updating then
+    require("lvim.impatient").clear_cache()
   end
 
   require("lvim.config"):init()
 
   require("lvim.plugin-loader").init {
     package_root = self.pack_dir,
     install_path = self.packer_install_dir,
+    updating = updating,
   }
 
+  if updating then
+    require("lvim.updater").init {
+      packer_cache_path = self.packer_cache_path,
+      lua_cache_path = self.lua_cache_path,
+    }
+  end
+
   return self
 end
 
----Update LunarVim
----pulls the latest changes from github and, resets the startup cache
+---@deprecated
 function M:update()
-  require_clean("lvim.utils.hooks").run_pre_update()
-  require_clean("lvim.utils.git").update_base_lvim()
-  require_clean("lvim.utils.hooks").run_post_update()
+  vim.api.nvim_err_writeln "LvimUpdate has been deprecated. Exit lvim and use `lvim --update` instead"
```

Comment:
```suggestion
  local msg =
    "Deprecation notice: LvimUpdate is no longer supported. See https://github.com/LunarVim/LunarVim#breaking-changes"

  vim.schedule(function()
    require("lvim.core.log"):warn(msg)
  end)
```

Reply:
I don't see why this is better.

1. It doesn't state what the user should use instead, which extremely unhelpful.
2. Instead it requires the user to visit a URL which _does not_ have a permalink.
3. It unnecessarily uses logging for something which is direct feedback to a user action

Reply:
1. It's more of a pattern we've been using: https://github.com/LunarVim/LunarVim/blob/ec1e4140fa4156df50785da5d6297145762a7daa/lua/lvim/config/init.lua#L47-L60
2. I realize it's not a permalink, but it's intentional since it can be more flexible with the instructions
3. I don't really feel it needs to be an error, but rather a warning. And this amount of code is the same it would take for `vim.notify`, which is already handled directly in the logger

That said, we can still add a note, while still pointing to the README
> use `lvim --update` instead

Reply:
That pattern is terribly unhelpful for a command, in contrast to user settings. It also doesn't point out _which_ change in the breaking changes. This _immediately_ provides the correct information and _only_ the information they need. It only displays when the user tries, not when the user loads `lvim`.

For a user's _config_ I see the usefulness of the pattern because a setting is deprecated, of which there can be many, and it should be logged so that a user can review any series of them.